src/ActivitiesSDKAdapter.js

```diff
@@ -18,5 +22,58 @@ import {ActivitiesAdapter} from '@webex/component-adapter-interfaces';
 export default class ActivitiesSDKAdapter extends ActivitiesAdapter {
   constructor(datasource) {
     super(datasource);
+
+    this.activityObservables = {};
+  }
+
+  /**
+   * Returns a promise to activity data from Webex.
+   *
+   * @param {string} activityID ID of the activity for which to fetch data
+   * @returns {Promise.<Activity>} Information about the activity of the given ID
+   *
+   * @private
+   */
+  fetchActivity(activityID) {
+    logger.debug('ACTIVITY', activityID, 'fetchActivity()', ['called with', {activityID}]);
+
+    return this.datasource.messages.get(activityID);
+  }
+
+  /**
+   * Returns an observable that emits activity data of the given ID.
+   *
+   * @param {string} ID ID of activity to get
+   * @returns {external:Observable.<Activity>} Observable stream that emits activity data
+   */
+  getActivity(ID) {
+    logger.debug('ACTIVITY', ID, 'getActivity()', ['called with', {ID}]);
+
+    if (!(ID in this.activityObservables)) {
+      // use ReplaySubject cause we don't need to set an initial value
+      this.activityObservables[ID] = new ReplaySubject(1);
+
+      defer(() => this.fetchActivity(ID)).pipe(
+        map((sdkMessage) => ({
+          ID: sdkMessage.id,
+          roomID: sdkMessage.roomId,
+          text: sdkMessage.text,
+          personID: sdkMessage.personId,
+          created: sdkMessage.created,
+          data: sdkMessage.attachments[0] && sdkMessage.attachments[0].content,
```

Comment:
leave this as `attachments` instead of `data`

and add another field, `card`:

```js
card: attachments && atachments[0] && attachments[0].contentType === 'xxx' && attachments[0].content